src/core/Form/Select/SingleSelect/SingleSelect.tsx

```diff
@@ -164,6 +187,14 @@ class BaseSingleSelect<T> extends Component<
     return null;
   }
 
+  componentDidUpdate(prevProps: SingleSelectProps<T & SingleSelectData>): void {
+    if (JSON.stringify(this.props.items) !== JSON.stringify(prevProps.items)) {
+      this.setState({
+        computedItems: this.props.items,
+      });
+    }
+  }
+
```

Comment:
I know comparing object equality with stringify is maybe not the most elegant solution. But it works 🤷‍♂️

Reply:
I have a hunch that this might be problematic if we use `JSON.stringify` performance wise.

We might have a some similar case in existing components where the prop change is wanted to detect. Will try to find it.

Reply:
I searched around and couldn't also found any more elegant solution that fits the case at hand. 
I will suggest that we put it to beta and and test it more. Keeping eyes open if it causes any issues while testing more down the line.